first-module/src/main/java/firstmodule/config/KafkaProducerConfig.java

```diff
@@ -0,0 +1,68 @@
+package firstmodule.config;
+
+import firstmodule.domain.CommandDomain;
+import org.apache.kafka.clients.admin.NewTopic;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.kafka.core.DefaultKafkaProducerFactory;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.core.ProducerFactory;
+import org.springframework.kafka.support.serializer.JsonSerializer;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Configuration
+public class KafkaProducerConfig {
+    @Value("${spring.kafka.bootstrap-servers}")
+    private String bootstrapAddress;
+
+    @Bean
+    public ProducerFactory<String, CommandDomain> producerFactoryJson() {
+        Map<String, Object> cfg = new HashMap<>();
+        cfg.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapAddress);
+        cfg.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+        cfg.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, JsonSerializer.class);
+
+        return new DefaultKafkaProducerFactory<>(cfg);
+    }
+
+    @Bean
+    public KafkaTemplate<String, CommandDomain> kafkaTemplateJson() {
+        return new KafkaTemplate<>(producerFactoryJson());
+    }
+
+//    @Bean
+//    public ProducerFactory<String, Long> producerFactoryLong() {
+//        Map<String, Object> cfg = new HashMap<>();
+//        cfg.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapAddress);
+//        cfg.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+//        cfg.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, LongSerializer.class);
+//
+//        return new DefaultKafkaProducerFactory<>(cfg);
+//    }
+//
+//    @Bean
+//    public KafkaTemplate<String, Long> kafkaTemplateLong() {
+//        return new KafkaTemplate<>(producerFactoryLong());
+//    }
+
+
+    @Bean
+    public NewTopic add() {
```

Comment:
Can you please create a component diagram to illustrate services interactions? Similar to those in the article - https://dzone.com/articles/building-microservices-inter-process-communication-2. It can be automated using Graphviz as described here - https://articles.microservices.com/an-alternative-way-of-visualizing-microservice-architecture-837cbee575c1, not mandatory.